### PR TITLE
[MODULAR][READY]Adds a colourable Badged Beret to the Loadout.

### PR DESCRIPTION
**Pull request by theselfish**
## About The Pull Request

Alternative to https://github.com/Skyrat-SS13/Skyrat-tg/pull/18821

## How This Contributes To The Skyrat Roleplay Experience

As TG moves to remove berets from non-head and Sec jobs, people here wish to retain them.
This can be considered an alternative to manually readding them every PR/recreating them. 
This option has the additional bonus of more player customization.

Though technically these aren't mutually exclusive, this simply means less oversight/work.

Q: Wouldn't this mean that people could easily impersonate the headgear of a head?
A: Kind of? But, if an Officer or someone is fooled by something as simple as a beret called 'badged beret' then I hope they aren't trigger-happy, as they most likely would fall for a toy emag or armblade.

Q: Why did you make a deity ugly beret for the test image?
A: To show that it wasn't just an old Sec beret.

Q: Why did you spell something so poorly, are you ESL?
A: No, I made this PR at 6 in the morning.

Q: Why does the description say "The latest in Anti-Territorial Government fashion!"? Are you some sort of TG Anti?
A: No. Don't be stupid. I simply put that there as a tongue-in-cheek joke about the de-militarization TG is going through. No ill-will intended. 

I would add the ability to rename/re-description it, but that would make Q1 even more complicated. I'm not editing code just to allow the description alone to be changed.

## Proof of Testing

image

<details>
<summary>Screenshots/Videos</summary>
  
</details>

## Changelog

:cl:
add: Badged Berets are now in the vendors and loadout.
/:cl:

### CODE REVIEW

**Comment by theselfish** [comment]
I am too tired to figure out if those are related to this.

**Comment by theOOZ** [comment]
we were _this_ close to atmos techs finding out how nice the welding hardhats are

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm [hunk 1]
```diff
@@ -336,3 +336,12 @@
...
 	greyscale_config_worn_vox = /datum/greyscale_config/standalone_hood/worn/oldvox
+
+/obj/item/clothing/head/beret/badge
+	name = "badged beret"
+	desc = "The latest in Anti-Territorial Government fashion!"
+	icon_state = "beret_badge"
+	greyscale_config = /datum/greyscale_config/beret_badge
+	greyscale_config_worn = /datum/greyscale_config/beret_badge/worn
+	greyscale_colors = "#972A2A#EFEFEF"
+	flags_1 = IS_PLAYER_COLORABLE_1
```
Comment (theselfish) [inline modular_skyrat/modules/customization/modules/clothing/head/head.dm:347]:
```suggestion
	flags_1 = IS_PLAYER_COLORABLE_1
	supports_variations_flags = CLOTHING_SNOUTED_VARIATION_NO_NEW_ICON
```

**Comment by theselfish** [comment]
I forgot the vox/tesh.

**Comment by theselfish** [comment]
https://user-images.githubusercontent.com/102828457/213951225-51c47ae8-0983-464c-b5da-bbd0717fa57c.mp4

File: modular_skyrat/modules/vox_sprites/code/head.dm [hunk 2]
```diff
@@ -16,6 +16,9 @@
 /obj/item/clothing/head/beret
 	greyscale_config_worn_vox = /datum/greyscale_config/beret/worn/vox
 
+/obj/item/clothing/head/beret/badge
+	greyscale_config_worn_teshari = /datum/greyscale_config/beret_badge/worn/teshari
```
Comment (theselfish) [inline modular_skyrat/modules/vox_sprites/code/head.dm]:
```suggestion
	greyscale_config_worn_teshari = /datum/greyscale_config/beret_badge/worn/vox
```

File: modular_skyrat/modules/vox_sprites/code/head.dm [hunk 3]
```diff
@@ -16,6 +16,9 @@
 /obj/item/clothing/head/beret
 	greyscale_config_worn_vox = /datum/greyscale_config/beret/worn/vox
 
+/obj/item/clothing/head/beret/badge
+	greyscale_config_worn_teshari = /datum/greyscale_config/beret_badge/worn/vox
```
Comment (theselfish) [inline modular_skyrat/modules/vox_sprites/code/head.dm]:
```suggestion
	greyscale_config_worn_vox = /datum/greyscale_config/beret_badge/worn/vox
```

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm [hunk 4]
```diff
@@ -336,3 +336,13 @@
 	greyscale_config_worn_teshari = /datum/greyscale_config/standalone_hood/worn/teshari
 	greyscale_config_worn_better_vox = /datum/greyscale_config/standalone_hood/worn/newvox
 	greyscale_config_worn_vox = /datum/greyscale_config/standalone_hood/worn/oldvox
+
+/obj/item/clothing/head/beret/badge
+	name = "badged beret"
+	desc = "The latest in Anti-Territorial Government fashion!"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/clothing/head/head.dm]: I don't really like what this is implying.

File: modular_skyrat/modules/modular_vending/code/autodrobe.dm [hunk 5]
```diff
@@ -29,6 +29,7 @@
 				/obj/item/clothing/under/costume/skyrat/kamishimo = 5,
 				/obj/item/clothing/under/costume/skyrat/yukata = 5,
 				/obj/item/clothing/under/costume/skyrat/kimono = 5,
+				/obj/item/clothing/head/beret/badge = 5,
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_vending/code/autodrobe.dm:32]: Why the autodrobe and not just the clothesmate instead? It's not really a costume, it's a beret, should go with the beret in the clothesmate, I reckon.

**Review by GoldenAlpharex — DISMISSED** [review]

File: modular_skyrat/modules/modular_vending/code/autodrobe.dm (hunk 5 above)
Comment (theselfish) [reply modular_skyrat/modules/modular_vending/code/autodrobe.dm:32]: I swear that the beret is only in the autodrobe...

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm (hunk 4 above)
Comment (theselfish) [reply modular_skyrat/modules/customization/modules/clothing/head/head.dm]: The implication is that berets are the typical wear of rebels and anti-government militants (Che Guevara springs to mind.).

I also mentioned the TG part in the body 
(Q: Why does the description say "The latest in Anti-Territorial Government fashion!"? Are you some sort of TG Anti?
A: No. Don't be stupid. I simply put that there as a tongue-in-cheek joke about the de-militarization TG is going through. No ill-will intended.)

If it's that much of a dealbreaker I can change it.

File: modular_skyrat/modules/modular_vending/code/autodrobe.dm (hunk 5 above)
Comment (theselfish) [reply modular_skyrat/modules/modular_vending/code/autodrobe.dm:32]: I stand corrected. Fixing right now.

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm (hunk 4 above)
Comment (GoldenAlpharex) [reply modular_skyrat/modules/customization/modules/clothing/head/head.dm]: I'm just not a fan of that being the description. If it was a comment, I wouldn't really say too much about it, but it's just the kind of very useless description that means absolutely nothing to everyone 3 months after it was added.

Comment (theselfish) [reply modular_skyrat/modules/customization/modules/clothing/head/head.dm]: Very well. It's a beret. Not much to describe it as though.

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm [hunk 6]
```diff
@@ -337,6 +337,15 @@
 	greyscale_config_worn_better_vox = /datum/greyscale_config/standalone_hood/worn/newvox
 	greyscale_config_worn_vox = /datum/greyscale_config/standalone_hood/worn/oldvox
 
+/obj/item/clothing/head/beret/badge
+	name = "badged beret"
+	desc = "The latest in Anti-Territorial Government fashion!"
```
Comment (theselfish) [inline modular_skyrat/modules/customization/modules/clothing/head/head.dm]:
```suggestion
	desc = "A beret. With a badge. What do you want, a dissertation? It's a hat."
```

**Review by tf-4 — APPROVED** [review]
this is still larp but whatever

**Comment by tf-4** [comment]
and lo, tf4 did say, let the furries have their hats

**Pull request merged**